webui/react/src/components/ConfigPolicies.tsx

```diff
@@ -0,0 +1,148 @@
+import handleError from 'utils/error';
+
+interface Props {
+  workspaceId?: number;
+}
+
+type FormInputs = {
+  task: string;
+};
+
+const ConfigPolicies: React.FC<Props> = ({ workspaceId }: Props) => {
+  const confirm = useConfirm();
+  const { openToast } = useToast();
+  const { canModifyWorkspaceConfigPolicies, loading: rbacLoading } = usePermissions();
+  const [form] = Form.useForm<FormInputs>();
+
+  const [disabled, setDisabled] = useState(true);
+
+  const APPLY_MESSAGE = "You're about to apply these config policies to this workspace.";
+  const VIEW_MESSAGE = 'An admin applied these config policies to this workspace.';
+
+  const updatePolicies = async () => {
+    if (workspaceId) {
+      try {
+        await updateWorkspaceConfigPolicies({
+          configPolicies: form.getFieldValue('task'),
+          workloadType: 'NTSC',
+          workspaceId,
+        });
+        openToast({ title: 'Config policies updated' });
+      } catch (error) {
+        handleError(error);
+      }
+    }
+  };
+
+  const confirmApply = () => {
+    confirm({
+      content: (
+        <span>
+          This will impact{' '}
+          <strong>
+            <u>all</u>
+          </strong>{' '}
+          underlying projects and their experiments in this workspace.
+        </span>
+      ),
+      okText: 'Apply',
+      onConfirm: updatePolicies,
+      onError: handleError,
+      size: 'medium',
+      title: APPLY_MESSAGE,
+    });
+  };
+
+  const loadableTaskConfigPolicies: Loadable<string | undefined> = useAsync(async () => {
+    if (workspaceId) {
+      const response = await getWorkspaceConfigPolicies({
+        workloadType: 'NTSC',
+        workspaceId,
+      });
+      return response.configPolicies;
+    }
+    return NotLoaded;
+  }, [workspaceId]);
+
+  const initialTaskYAML = yaml.dump(loadableTaskConfigPolicies.getOrElse(undefined));
+
+  if (rbacLoading) return <Spinner spinning />;
+
+  return (
+    <Column>
+      <Row width="fill">
+        <div style={{ width: '100%' }}>
+          {canModifyWorkspaceConfigPolicies ? (
+            <Alert
+              action={
+                <Button disabled={disabled} onClick={confirmApply}>
+                  Apply
+                </Button>
+              }
+              message={APPLY_MESSAGE}
+              showIcon
+            />
+          ) : (
+            <Alert message={VIEW_MESSAGE} showIcon />
+          )}
+        </div>
+      </Row>
+      <Row width="fill">
+        <div style={{ width: '100%' }}>
+          <Form form={form}>
+            <Form.Item
+              name="task"
+              rules={[
+                {
+                  validator: (_, value) => {
+                    try {
+                      yaml.load(value);
+                      if (value === initialTaskYAML) {
+                        setDisabled(true);
+                      } else {
+                        setDisabled(false);
+                      }
```

Comment:
is this a pattern we use elsewhere? It'd make sense to instead move this logic out of the validator because it's really dependent on the validation state and value. We could instead use a memo instead of state.